smtp/init.lua

```diff
@@ -155,27 +162,65 @@ curl_mt = {
             if not body or not url or not from then
                 error('request(url, from, to, body [, options]])')
             end
-            local header = ''
+            local header
             local recipients = {}
             header = 'From: ' .. from .. '\r\n' ..
                      'To: ' .. add_recipients(recipients, to) .. '\r\n'
             if opts.cc then
                 header = header .. 'Cc: ' .. add_recipients(recipients, opts.cc) .. '\r\n'
             end
+            if opts.subject then
+                header = header .. 'Subject: ' .. opts.subject .. '\r\n'
+            end
             add_recipients(recipients, opts.bcc)
             if opts.headers and #opts.headers > 0 then
                 header = header .. table.concat(opts.headers, '\r\n') .. '\r\n'
             end
+            local content_type = 'Content-Type: ' .. (opts.content_type or 'text/plain') ..
+            '; charset=' .. (opts.charset or  'UTF-8') ..';\r\n'
+            
+            if not opts.attachments or #opts.attachments == 0 then
+                body = header .. content_type ..'\r\n' .. body
+            else
+                -- multipart content according to https://tools.ietf.org/html/rfc1341
+                local MULTIPART_CONTENT_TYPE = 'Content-Type: multipart/mixed; boundary=MULTIPART-MIXED-BOUNDARY;\r\n'
+                local MULTIPART_SEPARATOR = '\r\n--MULTIPART-MIXED-BOUNDARY\r\n'
+                local MULTIPART_END = '\r\n--MULTIPART-MIXED-BOUNDARY--\r\n'
+                local attachments = ''
 
-            local content_type = opts.content_type or 'text/plain'
-            local charset = opts.charset or 'UTF-8'
-            header = header .. 'Content-Type: ' .. content_type .. '; charset=' .. charset .. '\r\n'
+                for _, attachment in ipairs(opts.attachments) do
+                    local base64_encode = attachment.base64_encode or true
```

Comment:
`false or true` is `true`.

Reply:
Fixed. Also I improved test suite to test both three cases: base64_encode = nil, false, true